python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -956,6 +1040,24 @@ def fetch_row_counts(
         """
         return self.attach_metadata(_fetch_row_count_metadata)
 
+    @public
+    @experimental
+    def fetch_column_metadata(
+        self,
+    ) -> (
+        "DbtEventIterator[Union[Output, AssetMaterialization, AssetObservation, AssetCheckResult]]"
+    ):
+        """Experimental functionality which will fetch column metadata and build column lineage
+        for dbt models in a dbt run in a separate thread. This deferred metadata fetch may be
```

Comment:
Let's remove the "May be more efficient", since that's pretty subjective language. Let's just include information about what is actually happening:

It fetches column schema for a model's upstream dependencies, it caches column schema for upstream models, runs sqlglot to generate column lineage, etc.